1-development/post-rating/src/main/java/com/invoicexpress/postrating/service/PostService.java

```diff
@@ -0,0 +1,52 @@
+package com.invoicexpress.postrating.service;
+
+import com.invoicexpress.postrating.Enum.VoteEnum;
+import com.invoicexpress.postrating.entity.Post;
+import com.invoicexpress.postrating.exceptions.InvalidPostIdException;
+import com.invoicexpress.postrating.repository.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+
+@Service
+public class PostService {
+
+    @Autowired
+    private PostRepository postRepository;
+
+    public Post updatePostVotes(int post_id, VoteEnum vote) throws InvalidPostIdException {
+
+        Post post = postRepository.findById(post_id).orElse(null);
+        if (post == null){
+            throw new InvalidPostIdException();
+
+        }
+        if (vote == VoteEnum.UP_VOTE) {
+            post.setUpVotes(post.getUpVotes() + 1);
+        } else {
+            post.setDownVotes(post.getDownVotes() + 1);
+        }
+        return postRepository.save(post);
+    }
+
+    public double calculateScoreOfPost(Post post){
+        float total = post.getDownVotes() + post.getUpVotes();
+        float upVotesOutTotal = post.getUpVotes()/total;
+        float confidentiality = 1.96f;
+
+        double score = (upVotesOutTotal + confidentiality*confidentiality/(2*total) - confidentiality * Math.sqrt(upVotesOutTotal * (1-upVotesOutTotal)
+                +confidentiality*confidentiality/(4*total))/total) / (1+confidentiality*confidentiality/total);
+        return score;
+    }
+
+    public List<ScorePostDTO> getAllPosts(){
+        List<Post> listOfPosts = postRepository.findAll();
+        List<ScorePostDTO> list = new ArrayList<>();
```

Comment:
Can you explain why you chose the ArrayList?

Reply:
Yes, I used ArrayList to create a dynamic array of ScorePostDTO and also I can be initialized

Reply:
Ok, I like the ArrayList :) but you are sorting the array. Can you use other List, Hash, Set (any sorted structure)?

PS: check the other question(about the equal, hash and comparable) :)

Reply:
If I implement the comparable I can use then the TreeSet because we will be sorting after each insertion then I don't need to sort the ArrayList and will reduce the complexity.

Reply:
👍